packages/cli/src/metadataGeneration/extension.ts

```diff
@@ -39,18 +44,19 @@ export function getExtensionsFromJSDocComments(comments: string[]): Tsoa.Extensi
     if (extensionData) {
       const keys = Object.keys(extensionData);
       keys.forEach(key => {
-        validateExtensionKey(key);
-
-        extensions.push({ key: key, value: extensionData[key] });
+        if (validateExtensionKey(key)) {
+          extensions.push({ key: key, value: extensionData[key] });
+        }
       });
     }
   });
 
   return extensions;
 }
 
-function validateExtensionKey(key: string) {
-  if (key.indexOf('x-') !== 0) {
+function validateExtensionKey(key: string): key is `x-${string}` {
```

Comment:
I would prefer 
```ts
asserts key is `x-${string}`
```
This is more intuitive when throwing and avoids the throwing `if/then/else`.

Also gives rise to renaming this function to `assertValidExtensionKey`

Reply:
Cool, didn't know about that TS feature – it definitely looks better!